### PR TITLE
[MJAVADOC-807] Simplify IT for MJAVADOC-498

### PR DESCRIPTION
**Pull request by michael-o**
This closes #309

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MJAVADOC) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue.  Your pull request should address just this issue, without
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MJAVADOC-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MJAVADOC-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify -Prun-its` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/it/projects/MJAVADOC-498_mm_modulepath/invoker.properties [hunk 1]
```diff
@@ -16,3 +16,4 @@
 # under the License.
 
 invoker.java.version = 9+
+invoker.goals=clean compile javadoc:javadoc
```
Comment (slawekjaranowski) [inline src/it/projects/MJAVADOC-498_mm_modulepath/invoker.properties:19]: clean is not necessary - invoker always clear project directory - https://maven.apache.org/plugins/maven-invoker-plugin/integration-test-mojo.html#cloneClean

File: src/it/projects/MJAVADOC-498_modulepath/verify.groovy [hunk 2]
```diff
@@ -17,5 +17,6 @@
  * under the License.
  */
 def log = new File( basedir, 'build.log').text
-
-assert log.count( " --module-path" ) == 3
+def count = log.count( " --module-path" )
+// depends on Maven version
+assert count == 2 || count == 3
```
Comment (slawekjaranowski) [inline src/it/projects/MJAVADOC-498_modulepath/verify.groovy:22]: We have a `mavenVersion` property defined so we can add something like 
```
assert mavenVersion.startWith('4') && count == x || mavenVersion.startWith('3') && count == y
```

**Review by slawekjaranowski — APPROVED** [review]

File: src/it/projects/MJAVADOC-498_mm_modulepath/invoker.properties (hunk 1 above)
Comment (michael-o) [reply src/it/projects/MJAVADOC-498_mm_modulepath/invoker.properties:19]: Awesome, didn't know. This is done in many ITs actually. Waste of cycles. Will change.

File: src/it/projects/MJAVADOC-498_modulepath/verify.groovy (hunk 2 above)
Comment (michael-o) [reply src/it/projects/MJAVADOC-498_modulepath/verify.groovy:22]: OK, let me try that. But how to solve before 3.9.x and after? That is a mere string.

Comment (slawekjaranowski) [reply src/it/projects/MJAVADOC-498_modulepath/verify.groovy:22]: yes it is a string

**Review by slawekjaranowski — APPROVED** [review]

File: src/it/projects/MJAVADOC-498_modulepath/verify.groovy (hunk 2 above)
Comment (michael-o) [reply src/it/projects/MJAVADOC-498_modulepath/verify.groovy:22]: I improved the comment and will leave as-is. Without a selector on an object that is too much fiddling.

**Pull request closed without merging**